plugins/rh-rbac-backend/src/service/policy-builder.ts

```diff
@@ -68,27 +110,308 @@ export class PolicyBuilder {
     router.use(permissionsIntegrationRouter);
 
     router.get('/', async (request, response) => {
-      const token = getBearerTokenFromAuthorizationHeader(
-        request.header('authorization'),
-      );
+      const decision = await authorize(env.identity, request, permissions, {
+        permission: policyEntityReadPermission,
+      });
 
-      const decision = (
-        await permissions.authorizeConditional(
-          [{ permission: policyEntityReadPermission }],
-          {
-            token,
-          },
-        )
-      )[0];
+      if (decision.result === AuthorizeResult.DENY) {
+        throw new NotAllowedError(); // 403
+      }
+      response.send({ status: 'Authorized' });
+    });
+
+    router.get('/policies', async (req, response) => {
+      const decision = await authorize(env.identity, req, permissions, {
+        permission: policyEntityReadPermission,
+      });
 
       if (decision.result === AuthorizeResult.DENY) {
-        response.status(403);
-        response.send({ status: 'Unauthorized' });
+        throw new NotAllowedError(); // 403
+      }
+
+      let policies: string[][];
+      if (isPolicyFilterEnabled(req)) {
+        const entityRef = getFirstQuery(req.query.entityRef);
+        const permission = getFirstQuery(req.query.permission);
+        const policy = getFirstQuery(req.query.policy);
+        const effect = getFirstQuery(req.query.effect);
+
+        const filter: string[] = [entityRef, permission, policy, effect];
+        policies = await enforcer.getFilteredPolicy(0, ...filter);
       } else {
-        response.send({ status: 'Authorized' });
+        policies = await enforcer.getPolicy();
       }
+
+      response.json(transformPolicyArray(...policies));
+    });
+
+    router.get('/policy/:namespace/:id', async (request, response) => {
```

Comment:
Should we move all the router.* implemenatations to its own file?

Reply:
Done.